gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/ProxyableDatasetVersionPredicate.java

```diff
@@ -0,0 +1,40 @@
+package gobblin.data.management.retention.policy;
+
+import java.util.Properties;
+
+import com.google.common.base.Predicate;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import org.apache.hadoop.fs.Path;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.data.management.retention.version.DatasetVersion;
+import gobblin.data.management.retention.version.StringDatasetVersion;
+import gobblin.util.ProxiedFileSystemUtils;
+
+
+@AllArgsConstructor
+public class ProxyableDatasetVersionPredicate implements Predicate<DatasetVersion> {
```

Comment:
Agree is it hacky, unfortunately I don't see a better short term solution to the problem. Will move to an internal repo.